backend/src/routes/spotify.js

```diff
@@ -78,18 +75,20 @@ async function handleOauthCallback(req, res) {
       return email;
     });
 
-    await Spotify.authorizationCodeGrant(code, (error, data) => {
+    await Spotify.authorizationCodeGrant(code, async (error, data) => {
       if (error) {
         throw error;
       }
       Spotify.setAccessToken(data.body.access_token);
+      const { body: spotifyProfile } = await Spotify.getMe();
 
       // We have a Spotify refresh token now.
       const refreshToken = data.body.refresh_token;
 
       // Get spotify data and store in the user's document
       db.collection('user_metadata').doc(userEmail).update({
         spotify_refresh_token: refreshToken,
+        spotify_profile: spotifyProfile,
```

Comment:
Quick question: Do we have a feature that requires spotify profile data? I'm not sure where this will be used in our app.

Reply:
resolved offline